pkg/scheduler/framework/plugins/noderesources/fit.go

```diff
@@ -44,14 +45,17 @@ const (
 
 // Fit is a plugin that checks if a node has sufficient resources.
 type Fit struct {
-	ignoredResources sets.String
+	ignoredResources      sets.String
+	ignoredResourceGroups sets.String
 }
 
 // FitArgs holds the args that are used to configure the plugin.
 type FitArgs struct {
 	// IgnoredResources is the list of resources that NodeResources fit filter
 	// should ignore.
 	IgnoredResources []string `json:"ignoredResources,omitempty"`
+	// IgnoredResourceGroups defines the list of resource prefix that NodeResources fit filter should ignore.
+	IgnoredResourceGroups []string `json:"ignoredResourceGroups,omitempty"`
```

Comment:
Let's add some format validation for ignoredResourceGroups. Maybe there is some validation for label keys that you could reuse.

Reply:
@alculquicondor can you clarify how this struct is used and why it has json annotations? is this serialized as a user-visible API or does it deserialize API input from a user?

Reply:
Yes, it is deserialized from component config. Please see #87784

Reply:
Ok... spreading API-oriented code out into non-API packages makes it very easy to make breaking changes without realizing it. Anything that serializes or deserializes content from a config file or API request should be in an API-specific package and should be versioned.

Reply:
Agreed. Do you prefer that #87784 gets solved before this PR?